docs/formplayer.rst

```diff
@@ -0,0 +1,138 @@
+Formplayer in HQ
+================
+
+Theis documentation describes how `formplayer <https://github.com/dimagi/formplayer/>`_ fits into the larger
+CommCare system, especially how it relates to CommCare HQ development. For details on building, runnning, and
+contributing to formplayer, see the formplayer repository.
+
+What Is Formplayer?
+^^^^^^^^^^^^^^^^^^^
+
+Formplayer is a Java Spring Boot server that wraps `commcare-core <https://github.com/dimagi/commcare-core>`_
+and presents its main features as an HTTP API. CommCare HQ's Web Apps, App Preview, and SMS Forms features are
+built on top of it:
+
+* Web Apps is a single-page application, inlined into a CommCare HQ template, the provides a web UI backed by the formplayer API.
+* App Preview is essentially the same as web apps, but embedded as a cell-phone-shaped iframe within the App Builder.
+* SMS Forms serializes a form filling session over SMS in a question / answer sequence that is handled
+by the main HQ process, which hits formplayer's API to send answers and get the next question.
+
+Repository Overview
+^^^^^^^^^^^^^^^^^^^
+
+.. image:: images/formplayer_repo_overview.png
+
+* `commcare-android <https://github.com/dimagi/commcare-android>`_: The UI layer of CommCare mobile.
+* `commcare-core <https://github.com/dimagi/commcare-core>`_: The CommCare engine, this powers both CommCare mobile and formplayer. Mobile uses the ``master`` branch, while formplayer uses the ``formplayer`` branch. The two branches have a fairly small diff.
+* `formplayer <https://github.com/dimagi/formplayer>`_
+* `commcare-hq <https://github.com/dimagi/commcare-hq>`_: HQ hosts web apps and the processes that run SMS forms.
+
+
+Relevant Architectural Decisions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+While a full detailing of formplayer's architecture is beyond the scope of this document, a few architectural
+decisions are particularly useful for HQ devs who are new to fromplayer to understand.
+
+Sandboxes
++++++++++
+Sharing the commcare-core code between mobile and formplayer allows us to keep CommCare Android and web apps
+essentially compatible. However, because commcare-core was first written for mobile and formplayer is a farly
+thin wrapper around it,
+paradigms that make more sense on mobile than on the web. Mobile is offline-first, so submitting
```

Comment:
```suggestion
essentially compatible. However, because commcare-core was first written for mobile some of the
paradigms it uses make more sense on mobile than on the web. Mobile is offline-first, so submitting
```